(From http://gcc.gnu.org/ml/gcc-patches/2002-10/msg00186.html)

Thu Oct  3 21:33:52 CEST 2002  Jan Hubicka  <jh@suse.cz>
	* i386.h (CPP_SPECS): fix defines for -msse, -msse2, -mpentium2,3.


Index: i386.h
===================================================================
RCS file: /cvs/gcc/egcs/gcc/config/i386/i386.h,v
retrieving revision 1.243.2.8
diff -c -3 -p -r1.243.2.8 i386.h
*** gcc/gcc/config/i386/i386.h	15 Jul 2002 06:54:36 -0000	1.243.2.8
--- gcc/gcc/config/i386/i386.h	3 Oct 2002 18:57:49 -0000
*************** extern int ix86_arch;
*** 584,592 ****
  %{march=pentium-mmx:-D__i586 -D__i586__ -D__pentium -D__pentium__ \
    -D__pentium__mmx__ \
    %{!mcpu*:-D__tune_i586__ -D__tune_pentium__ -D__tune_pentium_mmx__}}\
! %{march=pentiumpro|march=i686:-D__i686 -D__i686__ \
    -D__pentiumpro -D__pentiumpro__ \
    %{!mcpu*:-D__tune_i686__ -D__tune_pentiumpro__ }}\
  %{march=k6:-D__k6 -D__k6__ %{!mcpu*:-D__tune_k6__ }}\
  %{march=k6-2:-D__k6 -D__k6__ -D__k6_2__ \
    %{!mcpu*:-D__tune_k6__ -D__tune_k6_2__ }}\
--- 584,596 ----
  %{march=pentium-mmx:-D__i586 -D__i586__ -D__pentium -D__pentium__ \
    -D__pentium__mmx__ \
    %{!mcpu*:-D__tune_i586__ -D__tune_pentium__ -D__tune_pentium_mmx__}}\
! %{march=pentiumpro|march=i686|march=pentium2|march=pentium3:-D__i686 -D__i686__ \
    -D__pentiumpro -D__pentiumpro__ \
    %{!mcpu*:-D__tune_i686__ -D__tune_pentiumpro__ }}\
+ %{march=march=pentium2|march=pentium3: -D__pentium2 -D__pentium2__\
+   %{!mcpu*:-D__tune_pentium2__ }}\
+ %{march=pentium3: -D__pentium3 -D__pentium3__\
+   %{!mcpu*:-D__tune_pentium3__ }}\
  %{march=k6:-D__k6 -D__k6__ %{!mcpu*:-D__tune_k6__ }}\
  %{march=k6-2:-D__k6 -D__k6__ -D__k6_2__ \
    %{!mcpu*:-D__tune_k6__ -D__tune_k6_2__ }}\
*************** extern int ix86_arch;
*** 601,607 ****
  %{m386|mcpu=i386:-D__tune_i386__ }\
  %{m486|mcpu=i486:-D__tune_i486__ }\
  %{mpentium|mcpu=pentium|mcpu=i586|mcpu=pentium-mmx:-D__tune_i586__ -D__tune_pentium__ }\
! %{mpentiumpro|mcpu=pentiumpro|mcpu=i686|cpu=pentium2|cpu=pentium3:-D__tune_i686__ \
  -D__tune_pentiumpro__ }\
  %{mcpu=k6|mcpu=k6-2|mcpu=k6-3:-D__tune_k6__ }\
  %{mcpu=athlon|mcpu=athlon-tbird|mcpu=athlon-4|mcpu=athlon-xp|mcpu=athlon-mp:\
--- 605,611 ----
  %{m386|mcpu=i386:-D__tune_i386__ }\
  %{m486|mcpu=i486:-D__tune_i486__ }\
  %{mpentium|mcpu=pentium|mcpu=i586|mcpu=pentium-mmx:-D__tune_i586__ -D__tune_pentium__ }\
! %{mpentiumpro|mcpu=pentiumpro|mcpu=i686|mcpu=pentium2|mcpu=pentium3:-D__tune_i686__ \
  -D__tune_pentiumpro__ }\
  %{mcpu=k6|mcpu=k6-2|mcpu=k6-3:-D__tune_k6__ }\
  %{mcpu=athlon|mcpu=athlon-tbird|mcpu=athlon-4|mcpu=athlon-xp|mcpu=athlon-mp:\
*************** extern int ix86_arch;
*** 609,625 ****
  %{mcpu=athlon-4|mcpu=athlon-xp|mcpu=athlon-mp:\
  -D__tune_athlon_sse__ }\
  %{mcpu=pentium4:-D__tune_pentium4__ }\
! %{march=athlon-xp|march=athlon-mp|march=pentium3|march=pentium4:\
  -D__SSE__ }\
  %{march=pentium-mmx|march=k6|march=k6-2|march=k6-3\
  |march=athlon|march=athlon-tbird|march=athlon-4|march=athlon-xp\
! |march=athlon-mp|march=pentium2|march=pentium3|march=pentium4: -D__MMX__ }\
  %{march=k6-2|march=k6-3\
  |march=athlon|march=athlon-tbird|march=athlon-4|march=athlon-xp\
! |march=athlon-mp: -D__3dNOW__ }\
  %{march=athlon|march=athlon-tbird|march=athlon-4|march=athlon-xp\
  |march=athlon-mp: -D__3dNOW_A__ }\
! %{march=pentium4: -D__SSE2__ }\
  %{!march*:%{!mcpu*:%{!m386:%{!m486:%{!mpentium*:%(cpp_cpu_default)}}}}}"
  
  #ifndef CPP_CPU_SPEC
--- 613,629 ----
  %{mcpu=athlon-4|mcpu=athlon-xp|mcpu=athlon-mp:\
  -D__tune_athlon_sse__ }\
  %{mcpu=pentium4:-D__tune_pentium4__ }\
! %{march=athlon-xp|march=athlon-mp|march=pentium3|march=pentium4|msse|msse2:\
  -D__SSE__ }\
  %{march=pentium-mmx|march=k6|march=k6-2|march=k6-3\
  |march=athlon|march=athlon-tbird|march=athlon-4|march=athlon-xp\
! |march=athlon-mp|march=pentium2|march=pentium3|march=pentium4|mmx|msse|m3dnow: -D__MMX__ }\
  %{march=k6-2|march=k6-3\
  |march=athlon|march=athlon-tbird|march=athlon-4|march=athlon-xp\
! |march=athlon-mp|m3dnow: -D__3dNOW__ }\
  %{march=athlon|march=athlon-tbird|march=athlon-4|march=athlon-xp\
  |march=athlon-mp: -D__3dNOW_A__ }\
! %{march=pentium4|msse2: -D__SSE2__ }\
  %{!march*:%{!mcpu*:%{!m386:%{!m486:%{!mpentium*:%(cpp_cpu_default)}}}}}"
  
  #ifndef CPP_CPU_SPEC


